### DISCUSSION TITLE
pyTermTk - a Spiced Up Cross Compatible TUI Library 🌶️.

### DISCUSSION THREAD

**Discussion by ceccopierangiolieugenio** [pyodide/pyodide#5130] [Show and tell]
Hi everyone!
Thank you for developing such amazing library.

My Name is Eugenio and I made [**pyTermTk**](https://github.com/ceccopierangiolieugenio/pyTermTk), a self contained TUI library [cross compatible](https://ceccopierangiolieugenio.github.io/pyTermTk/info/features/crosscompatible.html) with Windows,Linux and OSX.
Pyodide allowed me to design an [HTML5 exporter](https://ceccopierangiolieugenio.github.io/pyTermTk/sandbox/sandbox.html) that allow any app to transparently run entirely on a browser

Some HTML5 projects exported are:

- [the DumbPaintTool](https://ceccopierangiolieugenio.itch.io/dumb-paint-tool) - The Terminal ASCII Photoshop
- [7drl-2024](https://ceccopierangiolieugenio.itch.io/a-snake-on-a-plane) - A Snake🐍 on a Plane✈️ - The Roguelike
- [breakoutRL](https://ceccopierangiolieugenio.itch.io/breakoutrl) - Breakout the Roguelike
- Several [tutorials](https://ceccopierangiolieugenio.github.io/pyTermTk/tutorial/005-calculator.html) and [examples](https://ceccopierangiolieugenio.github.io/pyTermTk/tutorial/000-examples.html): 

and

- The [Sandbox](https://ceccopierangiolieugenio.github.io/pyTermTk/sandbox/sandbox.html)

https://github.com/user-attachments/assets/754f0208-c5d2-4c10-ba18-0f6dcf9747bd

**Comment by ryanking13** [comment]
Awesome. Thanks for sharing your project. I always wondered if we could run TUI libraries like Textual in Pyodide, and it seems like `pyTermTk` can be a good example.

**Reply by ceccopierangiolieugenio** [reply]
... or you can just use [pyTermTk](https://github.com/ceccopierangiolieugenio/pyTermTk) 😂
